### PR TITLE
Solve the EGL configure issue in RenderEngine

### PR DESCRIPTION
**Pull request by txu4**
JIRA: https://01.org/jira/browse/AIA-70

After upgrade to HWC2.0, the window surface format is assigned as
PIXEL_FORMAT_RGBA_8888 in DisplayDevice::DisplayDevice, while
RenderEngine::selectEGLConfig doesnt provide Alpha info, the driver
will provide RGBX instead, which will cause CTS failure in
android.display.cts.VirtualDisplayXXX

Signed-off-by: Xu,Randy <randy.xu@intel.com>

### CODE REVIEW

File: services/surfaceflinger/RenderEngine/RenderEngine.cpp [hunk 1]
```diff
@@ -380,6 +380,8 @@ static status_t selectEGLConfig(EGLDisplay display, EGLint format,
         attribs[EGL_RED_SIZE]                   = 8;
         attribs[EGL_GREEN_SIZE]                 = 8;
         attribs[EGL_BLUE_SIZE]                  = 8;
+        if(format == HAL_PIXEL_FORMAT_RGBA_8888)
```
Comment (kalyankondapally) [inline services/surfaceflinger/RenderEngine/RenderEngine.cpp]: Can we get this to AOSP tree ?

Comment (kalyankondapally) [inline services/surfaceflinger/RenderEngine/RenderEngine.cpp]: @tpalli

Comment (tpalli) [inline services/surfaceflinger/RenderEngine/RenderEngine.cpp]: I'm not sure if this is a fix or a workaround, maybe we should change hwc to assign different format instead?

If upstreaming then maybe should be along the lines of 'if_has_alpha(format)' (not hardcoding for particular format enum) and also following coding style.

Comment (txu4) [inline services/surfaceflinger/RenderEngine/RenderEngine.cpp]: This fix is specific to below code in DisplayDevice.cpp, not a workaround.

DisplayDevice::DisplayDevice(
        const sp<SurfaceFlinger>& flinger,
        DisplayType type,
        int32_t hwcId,
#ifndef USE_HWC2
        int format,
#endif
        bool isSecure,
        const wp<IBinder>& displayToken,
        const sp<DisplaySurface>& displaySurface,
        const sp<IGraphicBufferProducer>& producer,
        EGLConfig config)
...
#ifdef USE_HWC2
        config = RenderEngine::chooseEglConfig(display, PIXEL_FORMAT_RGBA_8888);
#else
        config = RenderEngine::chooseEglConfig(display, format);
#endif

Comment (tpalli) [inline services/surfaceflinger/RenderEngine/RenderEngine.cpp]: OK I did not realize the format is hardcoded in case of HWC2, change makes sense then.

**Comment by txu4** [comment]
Patch for upstream is submitted https://android-review.googlesource.com/#/c/312066/

**Comment by kalyankondapally** [comment]
Thanks, commit message is missing:
Bug: (Jira issue id), u can create a new issue for adding virtual display support.

Test: (What tests are expected to pass).

**Comment by txu4** [comment]
JIRA: https://01.org/jira/browse/AIA-70

After upgrade to HWC2.0, the window surface format is assigned as
PIXEL_FORMAT_RGBA_8888 in DisplayDevice::DisplayDevice, while
RenderEngine::selectEGLConfig doesnt provide Alpha info, the driver
will provide RGBX instead, which will cause CTS failure in
android.display.cts.VirtualDisplayXXX

Test: Pass CtsDisplayTestCases and have no regression in other CTS test.

Signed-off-by: Xu,Randy randy.xu@intel.com

**Comment by kalyankondapally** [comment]
@txu4 Nitpick: JIRA: https://01.org/jira/browse/AIA-70 should be below u commit summary. We usually have Bug: <link to bug> but Jira seems better. I will update our wiki.

**Comment by kalyankondapally** [comment]
WIki updated: https://github.com/android-ia/manifest/wiki/Contributions

**Comment by kalyankondapally** [comment]
Merged

**Pull request closed without merging**